multiaddr/multiaddr.py

```diff
@@ -37,16 +37,12 @@ def __init__(self, string_addr=None, bytes_addr=None):
 
         Only one of string_addr or bytes_addr may be set
         """
-        if string_addr is not None and bytes_addr is not None:
-            raise ValueError(
-                "Only one of 'string_addr' or 'bytes_addr' may be set")
-
-        if string_addr is not None:
+        if string_addr is not None and bytes_addr is None:
             self._bytes = string_to_bytes(string_addr)
-        elif bytes_addr is not None:
+        elif bytes_addr is not None and string_addr is None:
             self._bytes = bytes_addr
         else:
-            raise ValueError("Invalid address type, must be bytes or str")
+            raise ValueError("Invalid address type, must be bytes xor str")
```

Comment:
I don't like this wording change, it reads like the user is supposed to pass in `b'foo' ^ "bar"`. I think most people parse `or` as an exclusive or, anyway.

Really I ought to fix this constructor to only take a single parameter, but the way python handles bytes and strings makes it basically impossible.